packages/nextjs/src/middleware/robots-middleware.test.ts

```diff
@@ -0,0 +1,109 @@
+import { expect } from 'chai';
+import sinon from 'sinon';
+import { NextApiRequest, NextApiResponse } from 'next';
+import { RobotsMiddleware } from './robots-middleware';
+import { SitecoreClient } from '@sitecore-content-sdk/core/client';
+import { SiteInfo } from '@sitecore-content-sdk/core/site';
+
+describe('RobotsMiddleware', () => {
+  const sandbox = sinon.createSandbox();
+  let sitecoreClientStub: sinon.SinonStubbedInstance<SitecoreClient>;
+  let middleware: RobotsMiddleware;
+  let req: Partial<NextApiRequest>;
+  let res: Partial<NextApiResponse>;
+
+  const mockSiteInfo: SiteInfo = {
+    name: 'test-site',
+    hostName: 'example.com',
+    language: 'en',
+  };
+
+  beforeEach(() => {
+    sitecoreClientStub = sandbox.createStubInstance(SitecoreClient);
+
+    res = {
+      setHeader: sandbox.stub(),
+      send: sandbox.stub(),
+      status: sandbox.stub().returnsThis(),
+    };
+
+    req = {
+      headers: {
+        host: 'example.com',
+      },
+    };
+
+    middleware = new RobotsMiddleware((sitecoreClientStub as unknown) as SitecoreClient);
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  it('should set the content type header to text/plain', async () => {
+    sitecoreClientStub.resolveSite.returns(mockSiteInfo);
+    sitecoreClientStub.getRobots.resolves('User-agent: *\nDisallow: /');
+
+    await middleware.getHandler()(req as NextApiRequest, res as NextApiResponse);
+
+    expect(res.setHeader).to.have.been.calledWith('Content-Type', 'text/plain');
+  });
+
+  it('should call resolveSite with hostname', async () => {
```

Comment:
can you, please, add an extra test to cover the case when fallback hostname `localhost` is used?